server/http-server.c

```diff
@@ -364,6 +364,12 @@ check_permission (HttpServer *htp_server, const char *repo_id, const char *usern
 {
     PermInfo *perm_info = NULL;
 
+    if (strcmp(op, "upload") == 0) {
+        int status = seaf_repo_manager_get_repo_status(seaf->repo_mgr, repo_id);
+        if (status != REPO_STATUS_NORMAL)
+            return EVHTP_RES_FORBIDDEN;
+    }
+
```

Comment:
This check should be moved to line 375. Doing it here will query the database for each block/commit/fs upload.